pkg/handlers/authorization/handler.go

```diff
@@ -61,7 +60,7 @@ func (auth *authorizationHandler) Process(req *http.Request, context *handlers.R
 		log.Trace("Handling a request with token...")
 		rolesProjects, err := auth.cache.getRolesAndProjects(context.Token)
 		if err != nil {
-			return req, fmt.Errorf("Could not determine the user or their roles: %v", err)
+			return req, err
```

Comment:
Depending on the caller who checks this errors, you could keep the `fmt.Errorf` with `%w` instead of `%v`. This allows golang to check errors like this: https://blog.golang.org/go1.13-errors

Reply:
I don't think this plays nice with the code that returns the structured error